### PR TITLE
Feat/CS-41983 - Exporting the stack role mappings for either all the teams in the org or for specific team

### PR DESCRIPTION
**Pull request by cs-raj**

### CODE REVIEW

File: packages/contentstack-export-to-csv/src/util/index.js [hunk 1]
```diff
@@ -678,6 +683,220 @@ function wait(time) {
   });
 }
 
+function handleErrorMsg(err) {
+  cliux.print(`Error: ${err?.errorMessage ? err?.message: messageHandler.parse('CLI_EXPORT_CSV_API_FAILED')}`, { color: 'red' });
```
Comment (antonyagustine) [inline packages/contentstack-export-to-csv/src/util/index.js:687]: @cs-raj Can you recheck this condition once again?

This seems wrong code: `err?.errorMessage ? err?.message : ''`


I suppose it should look as follows: `cliux.print('Error: ${(err?.errorMessage || err?.message) ? err?.errorMessage || err?.message : messageHandler.parse('CLI_EXPORT_CSV_API_FAILED')}', { color: 'red' })`

Comment (cs-raj) [reply packages/contentstack-export-to-csv/src/util/index.js]: Corrected it @antonyagustine

File: packages/contentstack-export-to-csv/src/util/index.js [hunk 2]
```diff
@@ -823,12 +840,61 @@ async function getTeamsDetail(allTeamsData, organization ,teamUid) {
...
+  let stackRoleWithTeamData = [];
+  if (teamUid) {
+    const team = allTeamsData.filter((team) => team?.uid === teamUid)[0];
+    for (const stack of team?.stackRoleMapping) {
+      const roleData = await mapRoleWithTeams(managementAPIClient, stack, team?.name, team?.uid);
+      stackRoleWithTeamData.push(...roleData);
+    }
+  } else {
+    for (const team of allTeamsData) {
+      for (const stack of team?.stackRoleMapping) {
```
Comment (antonyagustine) [inline packages/contentstack-export-to-csv/src/util/index.js:858]: We may possibly get `undefined is not iterable (cannot read property Symbol(Symbol.iterator))` error

Try to handle it like `for (const stack of team?.stackRoleMapping ?? [])`

NOTE: The same case may also apply to the parent loop. Just review it once.

File: packages/contentstack-export-to-csv/src/util/index.js [hunk 3]
```diff
@@ -823,12 +840,61 @@ async function getTeamsDetail(allTeamsData, organization ,teamUid) {
...
+
     write(this, team.users, fileName, 'Team User details');
-  
   }
 }
 
+async function exportRoleMappings(managementAPIClient, allTeamsData, teamUid) {
+  let stackRoleWithTeamData = [];
+  if (teamUid) {
+    const team = allTeamsData.filter((team) => team?.uid === teamUid)[0];
```
[the remark this answers is not in the document]
Comment (cs-raj) [reply packages/contentstack-export-to-csv/src/util/index.js]: done

File: packages/contentstack-export-to-csv/src/util/index.js (hunk 2 above)
Comment (cs-raj) [reply packages/contentstack-export-to-csv/src/util/index.js]: Checked both and modified the code

**Review by aman19K — APPROVED** [review]

**Review by antonyagustine — APPROVED** [review]

**Pull request merged**